integration-tests/deployment/environment.go

```diff
@@ -35,22 +39,43 @@ type OffchainClient interface {
 	jobv1.JobServiceClient
 	nodev1.NodeServiceClient
 	csav1.CSAServiceClient
+	// sometimes we need to connect to the nodes directly to execute actions that are not supported by JD/FMS
+	NodeClients() []*client.ChainlinkK8sClient
 }
 
+// TODO: we should rename it EVM chain, as it's not generic at all
 type Chain struct {
 	// Selectors used as canonical chain identifier.
 	Selector uint64
 	Client   OnchainClient
 	// Note the Sign function can be abstract supporting a variety of key storage mechanisms (e.g. KMS etc).
-	DeployerKey *bind.TransactOpts
+	// Rename to something more universal, as these keys are not only for deploying, but also for other actions
+	Keys []*bind.TransactOpts
+	// Function to execute if transaction submission fails.
+	RetrySubmit func(tx *types.Transaction, err error) (*types.Transaction, error)
```

Comment:
This should be included in confirm func

Reply:
@AnieeG it cannot, if we want to support both cases:
* transaction was not submitted (e.g. base was below minimum)
* transaction was submitted

Second case can indeed be retried in `Confirm(), because we have it there, it's not `nil`. But if submission fails we never get to `Confirm()` function and that's why I am calling `RetrySubmit()`.

Reply:
in the first case, won't it error when we call the gethwrapper function, then the caller of the gethwrapper function can just retry there?